### ISSUE TITLE
DPE Inspector: Physical Sky component crashes the Editor if either of Fog Height properties is changed

### ISSUE THREAD

**Issue by LB-ArturZieba** [o3de/o3de#16722]
**Describe the bug**
Modifying either *Fog Top Height* or *Fog Bottom Height* property in the *Physical Sky* component, while `ed_enableDPE` is set to `true` causes the Editor to crash.

*Call Stack*:
```
Call Stack Trace:
25) 00007FF8506CDEA7 (KERNELBASE) : UnhandledExceptionFilter
24) 00007FF84FFCC888 (InProcessClient64) : (function-name not available)
23) 00007FF84FFCC4D1 (InProcessClient64) : (function-name not available)
22) 00007FF852DB5330 (ntdll) : memset
21) 00007FF852D9C876 (ntdll) : _C_specific_handler
20) 00007FF852DB221F (ntdll) : _chkstk
19) 00007FF852D614B4 (ntdll) : RtlRaiseException
18) 00007FF852DB0D2E (ntdll) : KiUserExceptionDispatcher
17) C:\o3de\Code\Framework\AzQtComponents\AzQtComponents\Components\Widgets\Slider.cpp (359) : AzQtComponents::Slider::ShowValueToolTip
16) C:\o3de-packages\packages\qt-5.15.2-rev7-windows\qt\include\QtCore\qobjectdefs_impl.h (443) : QtPrivate::QFunctorSlotObject<`AzQtComponents::SliderDouble::SliderDouble'::`2'::<lambda_1>,1,QtPrivate::List<int>,void>::impl
15) 00007FF81A7A09AA (Qt5Core) : QObject::qt_static_metacall
14) 00007FF81C702BE0 (Qt5Widgets) : QAbstractSlider::setValue
13) 00007FF81C702AAC (Qt5Widgets) : QAbstractSlider::setSliderPosition
12) 00007FF81C799037 (Qt5Widgets) : QSlider::mouseMoveEvent
11) 00007FF81C6578E6 (Qt5Widgets) : QWidget::event
10) 00007FF81C6349A1 (Qt5Widgets) : QApplicationPrivate::notify_helper
 9) 00007FF81C6328C0 (Qt5Widgets) : QApplication::notify
 8) 00007FF81A7817CB (Qt5Core) : QCoreApplication::notifyInternal2
 7) 00007FF81C635BB5 (Qt5Widgets) : QApplicationPrivate::sendMouseEvent
 6) 00007FF81C680662 (Qt5Widgets) : QSizePolicy::QSizePolicy
 5) 00007FF81C67E5FE (Qt5Widgets) : QSizePolicy::QSizePolicy
 4) 00007FF81C6349A1 (Qt5Widgets) : QApplicationPrivate::notify_helper
 3) 00007FF81C633A23 (Qt5Widgets) : QApplication::notify
 2) 00007FF81A7817CB (Qt5Core) : QCoreApplication::notifyInternal2
 1) 00007FF81B934915 (Qt5Gui) : QGuiApplicationPrivate::processMouseEvent
```

It should be noted that this does not occur if `ed_enableDPE` is set to `false`.

This issue occurs on both Windows and Linux.

**Steps to reproduce**
1. Execute `ed_enable 1` command in the *Console*.
2. Create an entity and add *Physical Sky* component to it.
3. Enable the *Enable Fog* property.
4. Modify either *Fog Top Height* or *Fog Bottom Height* property.

**Expected behavior**
Property values are changed.

**Actual behavior**
Editor crashes.

**Video, Editor.log, error.log, and error.dmp**

https://github.com/o3de/o3de/assets/86952082/4644262e-d9a1-4e60-9677-ed4b968d24db

[Editor.log](https://github.com/o3de/o3de/files/12619581/Editor.log)
[error.log](https://github.com/o3de/o3de/files/12619582/error.log)
[error.zip](https://github.com/o3de/o3de/files/12619583/error.zip) - contains `error.dmp` file.

**Found in Branch**
Stabilization/2310 (372b2b8)

**Commit ID from [o3de/o3de](https://github.com/o3de/o3de) Repository**
372b2b829a767a8b522b05acc4b27810dbf49ad7

**Desktop**
 - Device: PC
 - OS: Windows
 - Version 10
 - CPU AMD Ryzen 5 3600X 6-Core Processor
 - GPU AMD Radeon RX 5700 XT
 - Memory 16GB

**Comment by LB-ArturZieba** [comment]
This issue was also encountered on the *Level Gradient Modifier* on the Stabilization/2310 (9b90a24) build.

**Comment by LB-ArturZieba** [comment]
A similar crash was also observed on *Gradient Surface Tag Emitter*, *Mesh Surface Tag Emitter*, *PhysX Collider Surface Tag Emitter*, and *Shape Surface Tag Emitter* components when changing final element in Tag array properties on the Stabilization/2310 (9b90a24) build. However, due to the generated call stack not matching up:
```
Call Stack Trace:
 9) 00007FF8506CDEA7 (KERNELBASE) : UnhandledExceptionFilter
 8) 00007FF84FFCC888 (InProcessClient64) : (function-name not available)
 7) 00007FF84FFCC4D1 (InProcessClient64) : (function-name not available)
 6) 00007FF852DB5330 (ntdll) : memset
 5) 00007FF852D9C876 (ntdll) : _C_specific_handler
 4) 00007FF852DB221F (ntdll) : _chkstk
 3) 00007FF852D614B4 (ntdll) : RtlRaiseException
 2) 00007FF852DB0D2E (ntdll) : KiUserExceptionDispatcher
 1) 00000230398C8E40 (module-name not available) : (function-name not available)
```

I am unsure if this should be reported as a separate issue, or will this be fixed by the fix to this issue?

**Comment by michalpelka** [comment]
We suffer it as well on the o3d-extras ROS 2 gem `ROS2FrameComponent`, issue is here: https://github.com/o3de/o3de-extras/issues/525.

**Comment by LB-MateuszSzweda** [comment]
Please note that the Color pickers in _Tint_ dropdowns of the **Bloom** component were also affected by this issue and changing the color crashed the Editor but the fix from https://github.com/o3de/o3de/pull/16764 was confirmed to also fix it.

**Comment by LB-ArturZieba** [comment]
Issue verified as fixed on the Stabilization/2310 (d380ffa) branch.
